core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1341,6 +1343,8 @@ public View create() {
       Preconditions.checkState(
           null != defaultNamespace, "Cannot create view without specifying a default namespace");
 
+      this.schema = TypeUtil.assignFreshIds(schema, new AtomicInteger(0)::incrementAndGet);
```

Comment:
It looks like this is being done in the wrong place because we need to assign fresh IDs in both the REST catalog here and in `BaseMetastoreViewCatalog`. We don't want to create a situation where implementations need to know to assign fresh IDs.

For tables, the initial assignment is done in `TableMetadata.newTableMetadata` and the reassignment for replace happens in `TableMetadata.buildReplacement`. Views should have a similar approach, where creating a replacement `ViewMetadata` automatically handles the reassignment so callers don't need to know and do it manually.

Reply:
yes you're right. One of the issues was that there wasn't a straightforward way inside `ViewMetadata` of detecting  whether `ViewMetadata` is created the first time or replaced. I think we can actually use the `history` field (or alternatively introduce a new field) to detect whether the view is created/replaced. I've updated the code so that no additional changes outside of `ViewMetadata` are required (and also moved the tests now into `TestViewMetadata`